app/models/color_scheme.rb

```diff
@@ -572,6 +575,37 @@ def is_dark?
   def is_wcag?
     base_scheme_id == NAMES_TO_ID_MAP["WCAG"] || base_scheme_id == NAMES_TO_ID_MAP["WCAG Dark"]
   end
+
+  def diverge_from_remote
+    new_scheme = dup
+    new_scheme.colors = self.colors_hashes
+    new_scheme.via_wizard = false
+    new_scheme.user_selectable = false
+    new_scheme.base_scheme_id = nil
+    new_scheme.skip_publish = true
+    new_scheme.remote_copy = true
+
+    DistributedMutex.synchronize("color_scheme_fork_#{self.id}") do
+      self.reload
+      if self.base_scheme_id.blank?
+        new_scheme.save!
+        self.base_scheme_id = new_scheme.id
+        self.save!
+      end
```

Comment:
```suggestion
      self.transaction do
        self.reload
        if self.base_scheme_id.blank?
          new_scheme.save!
          self.base_scheme_id = new_scheme.id
          self.save!
        end
      end
```

I think we should wrap the two saves in a database transaction to guard against cases where the caller of the method may not be calling this method in a database transaction.